src/Input.js

```diff
@@ -0,0 +1,61 @@
+import {
+  ERROR_MESSAGE_CAR_NAME_DUPLICATION,
+  ERROR_MESSAGE_CAR_NAME_INVALID,
+  ERROR_MESSAGE_USER_INPUT,
+  INPUT_MESSAGE_CAR_NAMES,
+  INPUT_MESSAGE_TRY_COUNT,
+} from './lib/constants.js';
+import {
+  checkArrayAllUnique,
+  readLineAsync,
+  splitIntoArray,
+  validatePositiveInteger,
+} from './lib/utils.js';
+
+class Input {
+  static #CAR_NAME_REGEXP = /^\w{1,5}$/;
+
+  static #SEPARATOR = ',';
+
+  #rawCars;
+
+  #rawTryCount;
+
+  constructor() {
+    this.#rawCars = '';
+    this.#rawTryCount = '';
+  }
+
+  async getUserInput() {
+    try {
+      this.#rawCars = await readLineAsync(INPUT_MESSAGE_CAR_NAMES);
+      this.#rawTryCount = await readLineAsync(INPUT_MESSAGE_TRY_COUNT);
+    } catch (e) {
+      throw new Error(ERROR_MESSAGE_USER_INPUT);
+    }
+  }
+
+  parseCars() {
+    const carArray = splitIntoArray(this.#rawCars, Input.#SEPARATOR);
+    Input.#validateCarArray(carArray);
+    return carArray;
+  }
+
+  parseTryCount() {
+    const tryCount = +this.#rawTryCount;
+    validatePositiveInteger(tryCount);
+    return tryCount;
+  }
+
+  static #validateCarArray(carArray) {
+    const isAllCarValid = carArray.every(car =>
+      Input.#CAR_NAME_REGEXP.test(car),
```

Comment:
```suggestion
      this.#CAR_NAME_REGEXP.test(car),
```
이렇게 작성하지 않으신 이유가 있으신가요?

Reply:
`validateCarArray`에 작성했습니다 !

Reply:
제가 클래스를 잘 몰라서.. 혹시 `validateCarArray` 안에서는 Input의 private 필드를 불러올 때 `this`가 아닌 `Input.`으로 접근해야 하는 건가요? 
정적 메소드라서 그런건가요?

Reply:
앗, 맞습니다. `validateCarArray`메서드는 정적 메서드여서 `클래스이름.메서드이름`으로 접근해야합니다.

이 정적 메서드와 인스턴스 메서드의 차이점은 인스턴스를 만들지 않고도 접근할 수 있냐 vs 만들어야 접근할 수 있냐 입니다.